nuxeo-core/nuxeo-core-test/src/test/java/org/nuxeo/ecm/core/TestSQLRepositoryQuery.java

```diff
@@ -2295,6 +2293,69 @@ public void testSelectColumnsSameName() throws Exception {
         res.close();
     }
 
+    /**
+     * @since 9.3
+     */
+    @Test(expected = QueryParseException.class)
+    public void testUnsupportedColumnMappings() throws Exception {
+        //ecm:majorVersion is NOT supported
+        String query = "SELECT ecm:majorVersion FROM File";
+        IterableQueryResult res = session.queryAndFetch(query, "NXQL");
+    }
+
+    /**
+     * @since 9.3
+     */
+    @Test
+    public void testSelectColumnMappings() throws Exception {
+        String query;
+        IterableQueryResult res;
+        Map<String, Serializable> map;
+
+        DocumentModel file = new DocumentModelImpl("/", "testfileForColumns", "File");
+        file.setPropertyValue("dc:title", "special1");
+        String content = "I am feeling elastic.";
+        String filename = "rucontent.txt";
+        Blob blob1 = Blobs.createBlob(content);
+        blob1.setFilename(filename);
+        file.setPropertyValue("content", (Serializable) blob1);
+        file = session.createDocument(file);
+        file.checkIn(VersioningOption.MAJOR, "major version");
+        session.save();
+
+        query = "SELECT dc:title FROM File WHERE dc:title = 'special1' AND ecm:isVersion = 1";
+        res = session.queryAndFetch(query, "NXQL");
+        assertEquals(1, res.size());
+        map = res.iterator().next();
+        assertEquals("special1", map.get("dc:title"));
+
+        query = "SELECT uid:major_version FROM File WHERE dc:title = 'special1' AND ecm:isVersion = 1";
+        res = session.queryAndFetch(query, "NXQL");
+        assertEquals(1, res.size());
+        map = res.iterator().next();
+        assertEquals(1l, map.get("uid:major_version"));
```

Comment:
Please use `1L` instead of `1l`, it's much more readable. Personally I even tend to avoid boxing and be explicit here: `Long.valueOf(1)`